client/me/social-login/index.jsx

```diff
@@ -173,9 +175,12 @@ class SocialLogin extends Component {
 export default connect(
 	state => {
 		const currentUser = getCurrentUser( state );
+		const connections = currentUser.social_login || [];
+		const googleConnection = find( connections, { service: 'google' } );
 
 		return {
-			isUserConnectedToGoogle: currentUser && currentUser.social_signup_service === 'google',
+			socialConnectionEmail: get( googleConnection, 'service_user_email', '' ),
+			isUserConnectedToGoogle: get( googleConnection, 'service', '' ) === 'google',
```

Comment:
It looks like checking we have at least a connection should be enough here:

```javascript
isUserConnectedToGoogle: ! isEmpty( googleConnection ),
```

Reply:
Resolved